.github/workflows/package-publish.yml

```diff
@@ -17,58 +17,34 @@ jobs:
   publish:
     runs-on: [self-hosted, 1ES.Pool=release-pool, 1ES.ImageOverride=ubuntu-22.04]
     environment: ${{ inputs.channel }}
+    env:
+      PMC_CLI_BASE_URL: ${{ secrets.PMC_CLI_BASE_URL }}
+      PMC_CLI_MSAL_CLIENT_ID: ${{ secrets.PMC_CLI_MSAL_CLIENT_ID }}
+      PMC_CLI_MSAL_SCOPE: ${{ secrets.PMC_CLI_MSAL_SCOPE }}
+      PMC_CLI_MSAL_AUTHORITY: ${{ secrets.PMC_CLI_MSAL_AUTHORITY }}
+      PMC_CLI_MSAL_CERT_PATH: auth.pem
 
     steps:
+      - uses: actions/checkout@v3
+
       - uses: actions/download-artifact@v3
         with:
           name: ${{ inputs.artifact }}
           path: packages
 
-      - name: Download repoclient
+      - name: Download certs
         env:
           CLIENT_ID: ${{ secrets.CLIENT_ID }}
           CLIENT_SECRET: ${{ secrets.CLIENT_SECRET }}
           TENANT_ID: ${{ secrets.TENANT_ID }}
           AZURE_VAULT: ${{ secrets.AZURE_VAULT }}
-          REPOCLIENT_KEY: ${{ secrets.REPOCLIENT_KEY }}
+          PMC_MSAL_CERT_NAME: ${{ secrets.PMC_MSAL_CERT_NAME }}
         run: |
           az login --service-principal --username "$CLIENT_ID" --password "$CLIENT_SECRET" --tenant "$TENANT_ID"
+          az keyvault secret download --vault-name "$AZURE_VAULT" --name "$PMC_MSAL_CERT_NAME" --file "${{ github.workspace }}/auth.pem"
 
-          az storage blob download \
-            --auth-mode login \
-            --account-name osconfig \
-            -c azure-repoapi-client \
-            -n azure-repoapi-client_2.0.1_amd64.deb \
-            -f repoclient.deb
-
-      - name: Install repoclient
-        run: |
-          sudo apt-get install python3-adal --yes
-          sudo dpkg -i repoclient.deb
-          rm repoclient.deb
-
-      - name: Configure repoclient
-        env:
-          AAD_CLIENT_ID: ${{ secrets.AAD_CLIENT_ID }}
-          AAD_CLIENT_SECRET: ${{ secrets.AAD_CLIENT_SECRET }}
-        run: |
-          config=$(cat <<EOF
-          {
-              "server": "azure-apt-cat.cloudapp.net",
-              "port": "443",
-              "repositoryId": "osconfig",
-              "AADAuthorityUrl": "https://login.microsoftonline.com",
-              "AADTenant": "72f988bf-86f1-41af-91ab-2d7cd011db47",
-              "AADResource": "https://microsoft.onmicrosoft.com/945999e9-da09-4b5b-878f-b66c414602c0",
-              "AADClientId": "$AAD_CLIENT_ID",
-              "AADClientSecret": "$AAD_CLIENT_SECRET"
-          }
-          EOF
-          )
-
-          echo "$config" > config.json
-
-      - name: Publish
+      - name: Set target
+        id: target
         env:
           REPO_ID_ubuntu_1804: ${{ secrets.REPO_ID_BIONIC }}
```

Comment:
although this isn't part of your change, would be nice to also have these arguments baked into the pipeline instead of using the secrets, again GitOps and tracking FTW! 😂

Reply:
Are we sure these repo IDs are "public" I remember at some point thinking that they were private. If they can be public, I'm all for it!

Reply:
Actually I realized why these are secrets. Sort of a hack to allow us to specify different repo IDs based on the environment (insiders-fast vs prod).

Reply:
Changed the repo IDs to [GitHub Environment Variables](https://docs.github.com/en/actions/learn-github-actions/contexts#vars-context) 👍